In [9]:
from autogen_ext.models.openai import AzureOpenAIChatCompletionClient
from dotenv import load_dotenv
load_dotenv()
import os
from autogen_core.models import UserMessage

In [6]:
AZURE_CONFIG = {
    'api_key': os.getenv("AZURE_OPENAI_API_KEY"),
    'api_version': os.getenv("AZURE_OPENAI_API_VERSION"),
    'azure_endpoint': os.getenv("AZURE_OPENAI_ENDPOINT"),
    'deployment_name': os.getenv("AZURE_OPENAI_DEPLOYMENT")
}

In [8]:
llm = AzureOpenAIChatCompletionClient(
    api_key=AZURE_CONFIG["api_key"],
    model=AZURE_CONFIG["deployment_name"],
    api_version=AZURE_CONFIG["api_version"],
    azure_endpoint=AZURE_CONFIG["azure_endpoint"]
)

In [11]:
result=await llm.create([UserMessage(content="what is the capital of india?",source="user")])
result.content

'The capital of India is New Delhi.'

### CreateResult(finish_reason='stop', content='The capital of India is New Delhi.', usage=RequestUsage(prompt_tokens=15, completion_tokens=9), cached=False, logprobs=None, thought=None)

In [12]:
from autogen_agentchat.agents import AssistantAgent
def web_search(query:str) -> str:
    """find information on the web"""
    return "Autogen is a programming framework for building multi-agent applications"

agent=AssistantAgent(
    name="assistant",
    model_client=llm,
    tools=[web_search],
    system_message="Answer the user questions as per your knowledge and you can also use web search tool based on the user questions."
    
)

In [13]:
result = await agent.run(task="Find information on AutoGen")


In [14]:
result

TaskResult(messages=[TextMessage(id='0e5dec3f-b3cf-4837-af85-928484c635de', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 9, 14, 13, 4, 59, 257044, tzinfo=datetime.timezone.utc), content='Find information on AutoGen', type='TextMessage'), ToolCallRequestEvent(id='f024d1c6-2acb-41d3-9d0c-9f78bf89caba', source='assistant', models_usage=RequestUsage(prompt_tokens=77, completion_tokens=17), metadata={}, created_at=datetime.datetime(2025, 9, 14, 13, 5, 0, 695457, tzinfo=datetime.timezone.utc), content=[FunctionCall(id='call_dGtYzbIIVNFelKqlitU1kPgA', arguments='{"query":"AutoGen technology"}', name='web_search')], type='ToolCallRequestEvent'), ToolCallExecutionEvent(id='5b3c7857-b6bd-4e8c-a44a-5fd1f118a84e', source='assistant', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 9, 14, 13, 5, 0, 706148, tzinfo=datetime.timezone.utc), content=[FunctionExecutionResult(content='Autogen is a programming framework for building multi-agent applic

TaskResult(messages=[
    TextMessage(id='0e5dec3f-b3cf-4837-af85-928484c635de', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 9, 14, 13, 4, 59, 257044, tzinfo=datetime.timezone.utc), content='Find information on AutoGen', type='TextMessage'), 
    
    ToolCallRequestEvent(id='f024d1c6-2acb-41d3-9d0c-9f78bf89caba', source='assistant', models_usage=RequestUsage(prompt_tokens=77, completion_tokens=17), metadata={}, created_at=datetime.datetime(2025, 9, 14, 13, 5, 0, 695457, tzinfo=datetime.timezone.utc), content=[FunctionCall(id='call_dGtYzbIIVNFelKqlitU1kPgA', arguments='{"query":"AutoGen technology"}', name='web_search')], type='ToolCallRequestEvent'), 
    
    ToolCallExecutionEvent(id='5b3c7857-b6bd-4e8c-a44a-5fd1f118a84e', source='assistant', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 9, 14, 13, 5, 0, 706148, tzinfo=datetime.timezone.utc), content=[FunctionExecutionResult(content='Autogen is a programming framework for building multi-agent applications', name='web_search', call_id='call_dGtYzbIIVNFelKqlitU1kPgA', is_error=False)], type='ToolCallExecutionEvent'),
    
     ToolCallSummaryMessage(id='44edb411-d827-4063-88e5-265a6eb72633', source='assistant', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 9, 14, 13, 5, 0, 706148, tzinfo=datetime.timezone.utc), content='Autogen is a programming framework for building multi-agent applications', type='ToolCallSummaryMessage', tool_calls=[FunctionCall(id='call_dGtYzbIIVNFelKqlitU1kPgA', arguments='{"query":"AutoGen technology"}', name='web_search')], results=[FunctionExecutionResult(content='Autogen is a programming framework for building multi-agent applications', name='web_search', call_id='call_dGtYzbIIVNFelKqlitU1kPgA', is_error=False)])], stop_reason=None)